pkg/detectors/hashicorpvaultauth/hashicorpvaultauth.go

```diff
@@ -0,0 +1,192 @@
+								roleId, secretId, vaultUrl))
+					}
+				}
+				results = append(results, s1)
+			}
+		}
+	}
+	return results, nil
+}
+
+// extractRoleIds finds all potential role_ids in the data
+func extractRoleIds(data string) []string {
+	var roleIds []string
+	uniqueRoleIds := make(map[string]struct{})
+
+	for _, match := range roleIdPat.FindAllStringSubmatch(data, -1) {
+		if len(match) >= 2 {
+			roleId := strings.TrimSpace(match[1])
+			if _, exists := uniqueRoleIds[roleId]; !exists {
+				uniqueRoleIds[roleId] = struct{}{}
+				roleIds = append(roleIds, roleId)
+			}
+		}
+	}
+	return roleIds
+}
+
+// extractSecretIds finds all potential secret_ids in the data
+func extractSecretIds(data string) []string {
+	var secretIds []string
+	uniqueSecretIds := make(map[string]struct{})
+
+	for _, match := range secretIdPat.FindAllStringSubmatch(data, -1) {
+		if len(match) >= 2 {
+			secretId := strings.TrimSpace(match[1])
+			if _, exists := uniqueSecretIds[secretId]; !exists {
+				uniqueSecretIds[secretId] = struct{}{}
+				secretIds = append(secretIds, secretId)
+			}
+		}
+	}
+	return secretIds
+}
+
+// extractVaultUrls finds all potential Vault URLs in the data
+func extractVaultUrls(data string) []string {
+	var urls []string
+	uniqueUrls := make(map[string]struct{})
+
+	for _, match := range vaultUrlPat.FindAllString(data, -1) {
+		url := strings.TrimSpace(match)
+		if _, exists := uniqueUrls[url]; !exists {
+			uniqueUrls[url] = struct{}{}
+			urls = append(urls, url)
+		}
+	}
+	return urls
+}
+
+func verifyMatch(ctx context.Context, client *http.Client, roleId, secretId, vaultUrl string) (bool, map[string]string, error) {
+	payload := map[string]string{
+		"role_id":   roleId,
+		"secret_id": secretId,
+	}
+
+	jsonPayload, err := json.Marshal(payload)
+	if err != nil {
+		return false, nil, err
+	}
+
+	// Construct the AppRole login endpoint from the base Vault URL
+	loginEndpoint := strings.TrimSuffix(vaultUrl, "/") + "/v1/auth/approle/login"
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, loginEndpoint, bytes.NewBuffer(jsonPayload))
+	if err != nil {
+		return false, nil, err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("X-Vault-Namespace", "admin")
+
+	res, err := client.Do(req)
+	if err != nil {
+		return false, map[string]string{"endpoint": loginEndpoint}, err
+	}
+
+	defer func() {
+		_, _ = io.Copy(io.Discard, res.Body)
+		_ = res.Body.Close()
+	}()
+
+	extraData := map[string]string{"endpoint": loginEndpoint}
+
+	switch res.StatusCode {
+	case http.StatusOK:
+		return true, extraData, nil
+	case http.StatusBadRequest:
+		return false, extraData, nil
+	case http.StatusUnauthorized, http.StatusForbidden:
+		return false, extraData, nil
+	case http.StatusNotFound:
+		return false, extraData, nil
+	default:
+		extraData["error"] = fmt.Sprintf("unexpected_status_%d", res.StatusCode)
+		return false, extraData, fmt.Errorf("unexpected HTTP response status %d", res.StatusCode)
+	}
```

Comment:
I’m not certain that including the status code in the extra data adds much value, given that it’s already conveyed through the error return.